app/imports/ui/pages/CompanyListing.jsx

```diff
@@ -52,16 +27,9 @@ const CompanyListing = () => {
         </Col>
         <Col>
           <Row>
-            { sampleCompanies.map(({ image, name, description, link }, index) => (
-              <Col xs={12} md={4} key={index}>
-                <Card className="justify-content-center" id="listing-card">
-                  <Card.Img id="listing-card-image" variant="top" src={image} />
-                  <Card.Title id="listing-card-title">{name}</Card.Title>
-                  <Card.Text id="listing-card-text">{description}</Card.Text>
-                  <Card.Link id="listing-card-link" to={link}><Button className="visit-button">Visit</Button></Card.Link>
-                </Card>
-              </Col>
-            )) }
+            { ready ? (companies.map((company, id) => (
+              <Company company={company} key={id} />
+            ))) : (<LoadingSpinner />)}
```

Comment:
This should be:
```jsx
{ ready ? (companies.map((company) => (
  <Company company={company} key={company._id} />
))) : <LoadingSpinner /> }
```

The way you are doing it, id is just the index of the array, which is not an identifier of the company document.